src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1654,7 +1654,7 @@ trait Namers extends MethodSynthesis {
         if (sym hasAnnotation NativeAttr)
           sym resetFlag DEFERRED
         else {
-          if (!symbolAllowsDeferred && ownerRequiresConcrete) fail(AbstractVar)
+          if (!symbolAllowsDeferred && ownerRequiresConcrete && !sym.isJava) fail(AbstractVar)
```

Comment:
Ok, so after some more testing I can observe that this specific change is only required to errors that occur with java *interfaces*.

The typer changes are still required for all java constructs though.